File: 2023-elixir/test/day_12_test.exs
defmodule Day12Test do
  use ExUnit.Case

  import TestHelper
  import Day12

  test "part1" do
    assert part_1(input("day_12_example_1")) == 21
    assert part_1(input("day_12")) == 7260
  end

  test "part2" do
    # assert part_2(input("day_11")) == 613_686_987_427
  end
end

File: 2024/day12/day12_test.go
package day12

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestPart1(t *testing.T) {
	assert.Equal(t, 140, Part1("input_example_1"))
	assert.Equal(t, 772, Part1("input_example_2"))
	assert.Equal(t, 1930, Part1("input_example_3"))
	assert.Equal(t, 1488414, Part1("input"))
}

func TestPart2(t *testing.T) {
	assert.Equal(t, 80, Part2("input_example_1"))
	assert.Equal(t, 436, Part2("input_example_2"))
	assert.Equal(t, 1206, Part2("input_example_3"))
	assert.Equal(t, 911750, Part2("input"))
}
